README.rst

```diff
@@ -0,0 +1,325 @@
+``PurchaseWorkflow`` program and compute the total purchases made by the particular user
+and total purchases made for the particular product in a specific time range. We will use three MapReduce
+programs ``PurchaseEventParser``, ``PurchaseCounterByUser``, and ``PurchaseCounterByProduct`` to apply
+different processing on the purchase events and Workflow ``PurchaseWorkflow`` to connect these MapReduce
+programs in a desired fashion.
+
+The result of the Workflow execution is persisted into a Dataset; the
+application uses the ``createDataset`` method to define the Dataset. We use three datasets,
+``purchaseRecords`` to store the valid parsed purchase events, ``userPurchases`` to store the total purchases
+made by particular user, and ``productPurchases`` to store the total purchases made for per product.
+The ``Purchase`` class defines the type used to store the parsed purchase events.
+
+The application also adds a custom Workflow action ``EmailNotifier``. When Workflow executes a custom action,
+the ``run`` method in it gets invoked. In ``EmailNotifier``, we only add a log statement, however it can be
+customized to send emails to the concerned parties.
+
+The ``PurchaseWorkflow`` is scheduled to execute every one hour.
+
+Finally, the application adds a service for querying the results from the Dataset.
+
+Let's take a closer look at the Workflow program.
+
+The ``PurchaseWorkflow`` extends an `AbstractWorkflow
+<http://docs.cdap.io/cdap/current/en/reference-manual/javadocs/co/cask/cdap/api/workflow/AbstractWorkflow.html>`__
+class and overrides the ``configure()`` method:
+
+.. code:: java
+
+  public class PurchaseWorkflow extends AbstractWorkflow {
+    @Override
+    protected void configure() {
+      setName("PurchaseWorkflow");
+      setDescription("Workflow to parse the purchase events and count the revenue per user and per product");
+
+      addMapReduce("PurchaseEventParser");
+
+      condition(new PurchaseEventVerifier())
+        .fork()
+          .addMapReduce("PurchaseCounterByUser")
+        .also()
+          .addMapReduce("PurchaseCounterByProduct")
+        .join()
+      .otherwise()
+        .addAction(new NotifyByEmail())
+      .end();
+    }
+  }
+
+In the ``configure()`` method we specify the topology for connecting the programs which will run as a part of
+the Workflow execution. For ``PurchaseWorkflow``, after setting the name and description for the Workflow, we add
+MapReduce program ``PurchaseEventParser``. This program will parse the raw purchase events to create the ``Purchase``
+objects from it. After this, we add ``condition`` in the Workflow, which takes a predicate ``PurchaseEventVerifier``.
+If the predicate evaluates to true, we `fork` the execution of the Workflow into two parallel branches.
+One branch executes the ``PurchaseCounterByUser`` MapReduce program, while other executes the
+``PurchaseCounterByProduct`` MapReduce program. If the predicate evaluates to false, then the programs added
```

Comment:
Insert a blank line before, and then:
```
If the predicate evaluates to false, then programs added in the ``otherwise`` section will be executed. We have added a single dummy custom action, ``EmailNotifier()`` to the ``otherwise`` section as an example of what is possible.
```